drivers: kscan: increase init priority

gmarull opened the pull request:
Some kscan drivers, e.g. ft5336 will not work with Zephyr default
settings due to wrong init priority order (e.g. I2C init order is 60 and
kscan is 40). kscan priority order has been increased and some custom
priorities removed since they are no longer needed.

MaureenHelm reviewed (changes requested):
We lose some configurability with this change. It would be better to just increase the default defined in `drivers/kscan/Kconfig`

avisconti commented on drivers/kscan/Kconfig:
```diff
@@ -18,10 +18,4 @@ module = KSCAN
-	default 40
-	help
-	  Keyboard scan device driver initialization priority.
-
```
Why removing completely KSCAN_INIT_PRIORITY and not, for example, change its default to APPLICATION_INIT_PRIORITY? In this way it is possible to configure it if required by some application

gmarull replied:
changed value and removed some custom defaults that are no longer needed.

gmarull committed:
drivers: kscan: increase init priority

Make kscan init priority higher, since it is lower than some buses like
I2C, making some devices to fail initialization if not tuning
priorities.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: remove custom kscan priority for ft5336

kscan priority is now higher than I2C, so there is no need for a custom
priority.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

VenkatKotakonda commented on drivers/kscan/Kconfig:
```diff
@@ -20,7 +20,7 @@ source "subsys/logging/Kconfig.template.log_config"
 config KSCAN_INIT_PRIORITY
 	int "Keyboard scan driver init priority"
-	default 40
+	default 90
```
Kscan in independent of other buses like i2c, how changing priority impacts the init?
